examples/blend/main.go

```diff
@@ -0,0 +1,193 @@
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"image"
+	_ "image/png"
+	"log"
+	"math"
+
+	"golang.org/x/image/colornames"
+	"golang.org/x/image/font/inconsolata"
+
+	"github.com/hajimehoshi/ebiten/v2"
+	"github.com/hajimehoshi/ebiten/v2/examples/resources/images/blend"
+	"github.com/hajimehoshi/ebiten/v2/text"
+)
+
+const (
+	screenWidth  = 780
+	screenHeight = 600
+)
+
+// mode is a blend mode with description.
+type mode struct {
+	blend ebiten.Blend
+	name  string
+}
+
+// Game is a canvas for drawing blend mode tiles.
+type Game struct {
+	source       *ebiten.Image
+	dest         *ebiten.Image
+	offscreen    *ebiten.Image
+	tileSize     float64
+	gridW, gridH float64
+	tileSpacing  float64 // Horizontal and vertical margin between tiles.
+	textSpacing  float64 // Additional vertical margin for text.
+	modes        []mode
+}
+
+func NewGame() *Game {
+	g := &Game{
+		source: loadImage(blend.Source_png),
+		dest:   loadImage(blend.Dest_png),
+		gridW:  4, gridH: 3,
+		tileSpacing: 64, textSpacing: 16,
+	}
+
+	// Adding all known blend modes and their names.
+	g.modes = []mode{
+		{blend: ebiten.BlendCopy, name: "BlendCopy"},
+		{blend: ebiten.BlendSourceAtop, name: "BlendSourceAtop"},
+		{blend: ebiten.BlendSourceOver, name: "BlendSourceOver"},
+		{blend: ebiten.BlendSourceIn, name: "BlendSourceIn"},
+		{blend: ebiten.BlendSourceOut, name: "BlendSourceOut"},
+		{blend: ebiten.BlendDestination, name: "BlendDestination"},
+		{blend: ebiten.BlendDestinationAtop, name: "BlendDestinationAtop"},
+		{blend: ebiten.BlendDestinationOver, name: "BlendDestinationOver"},
+		{blend: ebiten.BlendDestinationIn, name: "BlendDestinationIn"},
+		{blend: ebiten.BlendDestinationOut, name: "BlendDestinationOut"},
+		{blend: ebiten.BlendClear, name: "BlendClear"},
+		{blend: ebiten.BlendXor, name: "BlendXor"},
+		{blend: ebiten.BlendLighter, name: "BlendLighter"},
+	}
+
+	// Adjusting the tile size for the available images.
+	g.tileSize = maxSide(g.source, g.dest)
+	g.offscreen = ebiten.NewImage(int(g.tileSize), int(g.tileSize))
+
+	return g
+}
+
+func (g *Game) Update() error {
+	return nil
+}
+
+func (g *Game) Draw(screen *ebiten.Image) {
+	screen.Fill(colornames.White)
+
+	// Get an offset for center alignment.
+	// Where sw, sh is the screen size in pixels.
+	// And gw, gh is the grid size in pixels.
+	sw, sh := float64(screen.Bounds().Dx()), float64(screen.Bounds().Dy())
+	gw, gh := g.gridW*(g.tileSize+g.tileSpacing)-g.tileSpacing, g.gridH*(g.tileSize+g.tileSpacing)
+	cx, cy := (sw-gw)/2, (sh-gh)/2
+
+	// Drawing a tilemap
+	for y := 0.; y < g.gridH; y++ {
+		for x := 0.; x < g.gridW; x++ {
+			px, py := x*(g.tileSize+g.tileSpacing), y*(g.tileSize+g.tileSpacing)
+			if y > 0 {
+				// Making a place for the text.
+				py += g.textSpacing * y
+			}
+
+			// Drawing the blend mode and it's name.
+			mode := getMode(g.modes, int(x), int(y), int(g.gridW), int(g.gridH))
```

Comment:
Instead of rendering `empty`, what about letting `getMode` return mode and a boolean and skipping rendering when getMode returns false?

Reply:
Fixed